app/database/items/functors.go

```diff
@@ -0,0 +1,31 @@
+package items
+
+import (
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+)
+
+// OnlyVisibleBy returns a subview of the visible items for the given user basing on the given view
+func OnlyVisibleBy(user database.AuthUser) database.Functor {
+	return func(context *database.Context) {
```

Comment:
> for a better understanding, probably it should not be called context. Actually, it is just the current query on which you want to apply your "filter"

In the future it will also contain parameters such as table aliases and columns to join with.

Reply:
more args to the function but the context would still be the same, wouldn't it?

Reply:
No. The `SoThat` accepts multiple functors as arguments at once. SoThat(func1, func2, func3, ...). These functions are called with shared context. This way they can set parameters for each other. ```SoThat(items.JoinUsers(), As("my_users"))```

Reply:
I think it should be ```
type Context struct {
   DB
   vars map[string]interface{}
}```